lib/utils/public-key-util.js

```diff
@@ -26,15 +26,15 @@ var eventEmitter = new events.EventEmitter();
 
 module.exports = (function () {
 	logger.debug("Initializing");
-	var publicKeysJson;
+	var publicKeysJson = {};
 	var publicKeysEndpoint;
 	var isUpdateRequestPending = false;
 	
 	function setPublicKeysEndpoint(serverUrl) {
 		publicKeysEndpoint = serverUrl + PUBLIC_KEYS_PATH;
 	}
 	function getPublicKeysEndpoint() {
-		return publicKeysEndpoint;
+		return (arguments.length > 0) ? arguments[0] + PUBLIC_KEYS_PATH : publicKeysEndpoint;
```

Comment:
In a few places you're using the javascript function arguments field, which makes the function usage misleading. 

Can you be a bit more explicit using either function arguments or multiple functions in each cases